cpp/src/neighbors/ivf_pq/ivf_pq_search.cuh

```diff
@@ -137,6 +141,18 @@ void select_clusters(raft::resources const& handle,
       alpha = -1.0;
       beta  = 0.0;
     } break;
+    case cuvs::distance::DistanceType::CosineExpanded: {
```

Comment:
As @achirkin noted, the norms of the centers and of the queries should be accounted for when computing the cosine distance. Right now only the norms of the queries is used, and this can result in the wrong clusters getting selected.
In IVF-Flat: https://github.com/rapidsai/cuvs/blob/branch-24.10/cpp/src/neighbors/ivf_flat/ivf_flat_search.cuh#L166

Reply:
I tried to do that, but that gives poorer recall. Simply using inner product to select the clusters to probe gives better recall in the tests.